eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/client/EclairRpcClient.scala

```diff
@@ -568,4 +667,47 @@ class EclairRpcClient(val instance: EclairInstance)(
   def stop(): Option[Unit] = {
     process.map(_.destroy())
   }
+
+  /**
+    * Pings eclair every second to see if a invoice has been paid
+    * If the invoice has been paid or the payment has failed, we publish a
+    * [[org.bitcoins.eclair.rpc.json.PaymentResult PaymentResult]]
+    * event to the [[akka.actor.ActorSystem ActorSystem]]'s
+    * [[akka.event.EventStream ActorSystem.eventStream]]
+    */
+  override def monitorSentPayment(
+      paymentId: PaymentId, notifyPayment: PaymentResult => Future[Unit] = notifySentPayment): Future[PaymentResult] = {
+    val p: Promise[PaymentResult] = Promise[PaymentResult]()
+
+    val runnable = new Runnable() {
+
+      private var attempt = new AtomicInteger(1)
+
+      override def run(): Unit = {
+        for {
+          results <- getSentInfo(paymentId)
+        } yield {
+          results.find(_.status != PaymentStatus.PENDING) match {
+            case Some(result) =>
+              notifyPayment(result).foreach(_ => p.success(result))
+            case None =>
+              if (attempt.incrementAndGet() > 60) {
+                p.failure(new RuntimeException(
```

Comment:
Does this actually fail the payment in eclair itself? I feel like we shouldn't fail the payment attempt until eclair fails it internally. Shouldn't we do something with
 ```scala 
results.find(_.status == PaymentStatus.FAILED)`  match { ... } 
```

Reply:
We don't want to wait forever if Eclair doesn't respond for some reason.

Reply:
This seems like something that the user of the library should determine? Maybe this is some sort of a config?